### PR TITLE
Statbot input reading

### PR DESCRIPTION
**Pull request by sabinem**

### CODE REVIEW

File: R/read_data.R
Comment (sabinem) [inline R/read_data.R:1]: In this code `ds$read_path` was discontinued. Instead the format is used for deciding how to access the data.

File: R/create_dataset.R [hunk 1]
```diff
@@ -19,28 +19,25 @@
 #' @export
 create_dataset <- function(id) {
 
-  # the sheet is expected as a list in order to turn the row of the pipeline into a list
   datasets <- load_dataset_list()
-
   # turn the pipeline row into a list
-  ds_list <- datasets %>%
-    dplyr::filter(data_indicator == id) %>%
-    dplyr::mutate(sheet = as.list(sheet)) %>%
```
[the remark this answers is not in the document]
Comment (cmdoret) [reply R/create_dataset.R:28]: Looks better this way indeed

File: R/postgres_db_metadata.R [hunk 2]
```diff
@@ -57,7 +57,8 @@ list_tables_with_metadata_in_statbot_db <- function() {
   metadata <- get_metadata_from_postgres()
   tables_with_table_metadata <- metadata$metadata_tables$name
   tables_with_column_metadata <- unique(metadata$metadata_table_columns$table_name)
-  tables_with_metadata <- c(tables_with_table_metadata, tables_with_column_metadata)
+  tables_with_metadata <- c(tables_with_table_metadata, tables_with_column_metadata) %>%
```
Comment (cmdoret) [inline R/postgres_db_metadata.R:60]: Why did you add `unique()` here?

File: R/read_data.R [hunk 3]
```diff
@@ -56,18 +49,24 @@ read_data.default <- function(ds) {
...
+      method = download_method,
+      mode = "wb"
+    )
   )
   # Import the data
-  ds$data <- rio::import(temp_file, which = ds$sheet, header = TRUE)
+  if (ds$format == "xlsx") {
+    ds$data <- rio::import(temp_file)
+  } else {
+    ds$data <- rio::import(temp_file, which = ds$sheet, header = TRUE)
```
Comment (cmdoret) [inline R/read_data.R:65]: If the `sheet` field was removed from the dataset table, how is it derived now?
I assume it is derived for some xls datasets

**Review by cmdoret — COMMENTED** [review]
I agree this PR streamlines improves the dataset structure, but I'm worried it may break stuff.
Before merging it, I think it would be good to test loading one dataset from each type (px, csv, xls, json, rdf).

File: R/postgres_db_metadata.R (hunk 2 above)
Comment (sabinem) [reply R/postgres_db_metadata.R:60]: because most tables have metadata in both `tables_with_table_metadata` and  `tables_with_column_metadata` so the result was that I had most table_names in there double.

File: R/read_data.R (hunk 3 above)
Comment (sabinem) [reply R/read_data.R:65]: The `sheet` was not really needed. You can take a look here: https://github.com/statistikZH/statbotData/blob/main/data/const/statbot_input_data.csv

It contained mostly the `px_id` and for one dataset a text `data` : these were not needed as there is already another property `px_id` that contains the `px_id`.

Comment (cmdoret) [reply R/read_data.R:65]: Makes sense, but then line 65 above still uses `ds$sheet`. Shouldn't that be removed?

**Review by cmdoret — APPROVED** [review]
